concepts/vocabulary.md

```diff
@@ -0,0 +1,31 @@
+---
+grand_parent: Resources
+parent: Concepts
+title: Vocabulary
+description: The set of supported words
+---
+
+The **vocabulary** of a model is the set of supported words.
+In a machine translation system, the vocabulary may refer to the set of words supported in either the input or the output.
+A word that's supported in the output is a word that can be generated by the translation model.
+A word that's supported in the input is a word that can be translated.
+
+Typically the vocabulary is created from [training data](/customisation/training-data.md) by retaining the most frequent `N` words in the source and target language.
+Words that are not in the vocabulary are called out-of-vocabulary (OOV).
+
+The vocabulary of a language is the set of all possible words in the language.
+
+The vocabulary of a pretrained [word embedding model](/concepts/word-embeddings.md) is the set of all words with defined embeddings.
+
+## Challenges
+
+- A large vocabulary will allow the system to learn to translate more words, but makes the model much larger and slower.
+- In [neural machine translation](/approaches/neural-machine-translation.md), it can be challenging to share the vocabulary between the decoder inputs and decoder outputs.
```

Comment:
I would eliminate this sentence as it is hard to understand by non-developers/researchers.
Alternatively, we could further explain what sharing vocabulary between the inputs and outputs means and why it is done, but I feel it would make this article too granular.

Reply:
Yeah it would really expand the scope too much. That sort of thing belongs in the NMT article if anywhere